CuitService.Test/TaxInfoApi_ValidationTest.cs

```diff
@@ -0,0 +1,179 @@
+    {
+        private readonly WebApplicationFactory<Startup> _factory;
+        private readonly Fixture _specimenBuilders;
+
+        public TaxInfoApi_ValidationTest(WebApplicationFactory<Startup> factory)
+        {
+            _factory = factory;
+            _specimenBuilders = new Fixture();
+        }
+
+        [Theory]
+        [InlineData("20-31111111-8")]
+        [InlineData("20-31111111-6")]
+        [InlineData("20-31111111-1")]
+        public async Task GET_taxinfo_by_cuit_with_an_invalid_verification_digit_should_return_400_BadRequest(string cuit)
+        {
+            // Arrange
+            var client = _factory.WithBypassAuthorization().CreateClient();
+
+            // Act
+            var response = await client.GetAsync($"https://custom.domain.com/taxinfo/by-cuit/{cuit}");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+            // TODO: verify that the provider has not been called
+
+            var content = await response.Content.ReadAsStringAsync();
+            var problemDetail = JsonSerializer.Deserialize<JsonElement>(content);
+            Assert.Equal("One or more validation errors occurred.", problemDetail.GetProperty("title").GetString());
+            Assert.Collection(problemDetail.GetProperty("errors").EnumerateObject(),
+                item =>
+                {
+                    Assert.Equal("Cuit", item.Name);
+                    Assert.Equal(1, item.Value.GetArrayLength());
+                    Assert.Equal("The CUIT's verification digit is wrong", item.Value.EnumerateArray().First().GetString());
+                });
+        }
+
+        [Theory]
+        [InlineData("20-3111111-8")]
+        [InlineData("20-311111111-6")]
+        public async Task GET_taxinfo_by_cuit_with_wrong_length_should_return_400_BadRequest(string cuit)
+        {
+            // Arrange
+            var client = _factory.WithBypassAuthorization().CreateClient();
+
+            // Act
+            var response = await client.GetAsync($"https://custom.domain.com/taxinfo/by-cuit/{cuit}");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+            // TODO: verify that the provider has not been called
+
+            var content = await response.Content.ReadAsStringAsync();
+            var problemDetail = JsonSerializer.Deserialize<JsonElement>(content);
+            Assert.Equal("One or more validation errors occurred.", problemDetail.GetProperty("title").GetString());
+            Assert.Collection(problemDetail.GetProperty("errors").EnumerateObject(),
+                item =>
+                {
+                    Assert.Equal("Cuit", item.Name);
+                    Assert.Equal(1, item.Value.GetArrayLength());
+                    Assert.Equal("The CUIT number cannot have less than 11 numbers", item.Value.EnumerateArray().First().GetString());
+                });
+        }
```

Comment:
It could be an alternative...

```suggestion
            var problemDetail = JsonSerializer.Deserialize<ValidationProblemDetails>(content);
            // TODO: Add here assertion over typed data
            // . . . 
```